tlscanary/modes/performance.py

```diff
@@ -60,80 +70,143 @@ def run(self):
         self.save_profile(self.base_profile, "base_profile", log)
         log.stop(meta=meta)
 
+        # this will go away post-development
+        self.temp_debug_output()
+
+    # temporary debugging function
+    def temp_debug_output(self):
+        logging.info("Total change: %s" % self.total_change)
+        for i in xrange (0,3):
+            logging.info("Host: %s " % self.temp_hosts[i]["host"])
+            logging.info("Average change: %s " % self.temp_hosts[i]["connection_speed_change_average"])
+            logging.info("Median change: %s " % self.temp_hosts[i]["connection_speed_change_median"])
+            logging.info(self.temp_hosts[i]["connection_speeds"])
+            logging.info(self.temp_hosts[i]["connection_speeds_base"])
+
+    @staticmethod
+    def calculate_average_speed(sample_list):
+        num_samples = len(sample_list)
+        running_total = 0
+        for i in xrange(0, num_samples):
+            running_total += sample_list[i]
+        return float(running_total / num_samples)
+
+    @staticmethod
+    def calculate_median_speed(sample_list):
+        sample_list.sort()
+        num_samples = len(sample_list)
+        middle = num_samples/2
+        if num_samples % 2 == 1:
+            median = sample_list[middle]
+        else:
+            median = (sample_list[middle] + sample_list[middle-1]) / 2
+        return float(median)
+
     @staticmethod
-    def extract_connection_speed(uri_set):
-        new_set = []
-        for record in uri_set:
-            new_set.append([
-                record[0],  # URI rank
-                record[2].response.response_time - record[2].response.command_time  # connection speed
-            ])
-        return new_set
-
-    def consolidate_connection_speed_info(self, uri_set, connection_sets):
-        for record in uri_set:
-            temp_speeds = []
-            speed_aggregate = 0
-            for connection_set in connection_sets:
-                speed = [d for d in connection_set if d[0] == record[0]][0][1]
-                speed_aggregate += speed
-                temp_speeds.append(speed)
-            record[2].response.connection_speed_average = speed_aggregate / self.args.scans
-            record[2].response.connection_speed_samples = str(temp_speeds).strip('[]')
+    def calculate_speed_change(a, b):
+        change = (float(a - b) / b) * 100
+        format_to_two_decimals = float('%.2f' % change)
+        return format_to_two_decimals
+
+    def delete_last_log(self):
+        argv = self.args
+        argv.action = "delete"
+        argv.really = True
+        argv.include = ['1']
+        argv.exclude = []
+        log_delete = LogMode(argv, self.module_dir, self.tmp_dir)
+        log_delete.setup()
+        log_delete.run()
+
+    def extract_connection_speed(self, host_list, array_name):
+        for j in xrange(0, self.args.limit):
+            connection_speed = host_list[j]["response"]["response_time"] - host_list[j]["response"]["command_time"]
```

Comment:
You're using the right timestamps: `command_time` and `response_time` are sampled in JS when the command is received and the response is generated. We could move a bit closer to the query and its callbacks, but I doubt it would make much of a difference.